File: python/quicksort/function.py
def quicksort(array: list[int]) -> list[int]:
    if len(array) <= 1:
        return array
    
    else:
        pivot = array[0]
        left = [x for x in array[1:] if x < pivot]
        right = [x for x in array[1:] if x >= pivot]
        return quicksort(left) + [pivot] + quicksort(right)

File: javascript/quicksort/function.js
function quicksort(array) {
    if (array.length <= 1) {
        return array;
    } else {
        const pivot = array[0];
        const left = array.slice(1).filter(x => x < pivot);
        const right = array.slice(1).filter(x => x >= pivot);
        return [...quicksort(left), pivot, ...quicksort(right)];
    }
}
